src/main/java/org/sagebionetworks/web/client/widget/discussion/modal/NewThreadModal.java

```diff
@@ -30,19 +53,47 @@ public void hide() {
 
 	@Override
 	public void onSave() {
-		// TODO Auto-generated method stub
-		
+		String threadTitle = view.getTitle();
+		String messageMarkdown = view.getMessageMarkdown();
+		if (!isValidTitle(threadTitle)) {
+			synAlert.showError("Title cannot be empty.");
+			return;
+		}
+		if (!isValidMessage(messageMarkdown)) {
+			synAlert.showError("Message cannot be empty.");
+			return;
+		}
+		view.hideDialog();
+		CreateDiscussionThread toCreate = new CreateDiscussionThread();
+		toCreate.setForumId(forumId);
+		toCreate.setTitle(threadTitle);
+		toCreate.setMessageMarkdown(messageMarkdown);
+		discussionForumClient.createThread(toCreate, new AsyncCallback<DiscussionThreadBundle>(){
+			@Override
+			public void onFailure(Throwable caught) {
+				// TODO Auto-generated method stub
+				
+			}
+
+			@Override
+			public void onSuccess(DiscussionThreadBundle result) {
+				if (newThreadCallback != null) {
+					newThreadCallback.invoke(null);
```

Comment:
make sure you report success to the user via DisplayUtils.showInfo() (in a view)